independent-projects/bootstrap/core/src/main/java/io/quarkus/bootstrap/resolver/maven/MavenRepoInitializer.java

```diff
@@ -68,15 +67,17 @@
     private static final String MAVEN_HOME = "maven.home";
     private static final String M2_HOME = "M2_HOME";
     private static final String SETTINGS_XML = "settings.xml";
+    private static final String DASH_P = "-P";
```

Comment:
@aloubyansky wondering if you should scan for ` -P` instead, with a space. That would be a bit less fragile.

Reply:
Then -PprofileName won't work.

Reply:
@aloubyansky I'm still a bit worried about the fragility of this. I'm wondering if we could use the same machinery that Maven uses to parse the options? I don't know if it's something accessible and usable?

Reply:
I re-wrote it using API from maven-embedder (which is a new dependency).